contracts/schemes/SchemeFactory.sol

```diff
@@ -0,0 +1,172 @@
+pragma solidity ^0.5.16;
+
+import "@daostack/infra-experimental/contracts/votingMachines/IntVoteInterface.sol";
+import "@daostack/infra-experimental/contracts/votingMachines/VotingMachineCallbacksInterface.sol";
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "../utils/DAOFactory.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+
+
+/**
+ * @title A factory and registrar for Schemes for organizations
+ * @dev The SchemeFactory is used for deploying and registering schemes at organizations
+ */
+
+contract SchemeFactory is Initializable, VotingMachineCallbacks, ProposalExecuteInterface {
+    event NewSchemeProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _intVoteInterface,
+        string _schemeName,
+        bytes _schemeData,
+        uint64[3] _packageVersion,
+        bytes4 _permissions,
+        address _schemeToReplace,
+        string _descriptionHash
+    );
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _param);
+    event ProposalDeleted(address indexed _avatar, bytes32 indexed _proposalId);
+
+    // a SchemeProposal is a  proposal to add or remove a scheme to/from the an organization
+    struct SchemeProposal {
+        string schemeName;
+        bytes schemeData;
+        uint64[3] packageVersion;
+        address schemeToReplace;
+        bytes4 permissions;
+    }
+
+    mapping(bytes32=>SchemeProposal) public organizationProposals;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    DAOFactory public daoFactory;
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar this scheme referring to.
+     * @param _votingMachine the voting machines address to
+     * @param _voteParams voting machine parameters to register scheme.
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        DAOFactory _daoFactory
+    )
+    external
+    initializer
+    {
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteParams = _voteParams;
+        daoFactory = _daoFactory;
+    }
+
+    /**
+    * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision the voting result, 1 yes and 2 is no.
+    */
```

Comment:
@return ?

Reply:
There is no @return also in the other schemes documenting it, it either returns true if the function didn't fail and revert if it did fail, but it can only return true...

Reply:
so please add that to the function documentation :)

Reply:
ok